lib/src/widget/eh_gallery_list_card_.dart

```diff
@@ -258,4 +260,44 @@ class EHGalleryListCard extends StatelessWidget {
           decoration: gallery.isExpunged ? TextDecoration.lineThrough : null),
     );
   }
+
+
+  Widget _buildReadingProgress(BuildContext context) {
+    return GetBuilder<ReadProgressController>(
+      init: ReadProgressController.instance,
+      id: '${ReadProgressService.readProgressUpdateId}::${gallery.gid}',
+      builder: (_) {
+        return FutureBuilder<int>(
+          future: readProgressService.getReadProgress(gallery.gid),
+          builder: (context, snapshot) {
+            if (snapshot.connectionState != ConnectionState.done) {
+              return const SizedBox.shrink();
+            }
+
+            final readIndex = snapshot.data ?? 0;
+
+            double progress = gallery.pageCount != null && gallery.pageCount! > 0
+                ? ((readIndex + 1) / gallery.pageCount!).clamp(0.0, 1.0)
+                : 0.0;
+
+            // Don't show indicator if no progress
+            if (readIndex == 0.0) {
+              return const SizedBox.shrink();
+            }
+
+            return SizedBox(
+              width: 11,
+              height: 11,
+              child: CircularProgressIndicator(
+                value: progress,
+                strokeWidth: 2,
+                backgroundColor: UIConfig.galleryCardTextColor(context).withValues(alpha: 0.2),
```

Comment:
Are you using a newer version of Flutter?
<img width="2410" height="574" alt="image" src="https://github.com/user-attachments/assets/4c040f91-b307-49ba-b61c-7f696c0640bd" />